### PR TITLE
Docstrings for GraphDefinition

### PR DESCRIPTION
**Pull request by dpeng817**
Adds docstrings for public GraphDefinition methods

To enable https://github.com/dagster-io/dagster/pull/15078

### CODE REVIEW

**Comment by dpeng817** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15095** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15095" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15095?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/definitions/graph_definition.py [hunk 1]
```diff
@@ -367,16 +367,28 @@ def iterate_node_handles(
     @public
     @property
     def input_mappings(self) -> Sequence[InputMapping]:
+        """Input mappings for the graph.
+
+        An input mapping is a mapping from an input of the graph to an input of a child node.
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/definitions/graph_definition.py:372]: Appreciate the extra explanation

File: python_modules/dagster/dagster/_core/definitions/graph_definition.py [hunk 2]
```diff
@@ -367,16 +367,28 @@ def iterate_node_handles(
...
+        An output mapping is a mapping from an output of the graph to an output of a child node.
+        """
         return self._output_mappings
 
     @public
     @property
     def config_mapping(self) -> Optional[ConfigMapping]:
+        """The config mapping for the graph, if present.
+
+        By specifying a config mapping function, you can override the configuration for the child ops and graphs contained within a graph.
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/definitions/graph_definition.py:390]: nit: IMO reads smoother w/ "child ops and graphs" -> "child nodes"

File: python_modules/dagster/dagster/_core/definitions/graph_definition.py [hunk 3]
```diff
@@ -746,27 +758,72 @@ def get_resource_requirements(
...
+            @job
+            def do_it_all():
+                my_graph.alias("my_graph_alias")
+
+        """
         return super(GraphDefinition, self).alias(name)
 
     @public
     def tag(self, tags: Optional[Mapping[str, str]]) -> "PendingNodeInvocation":
+        """Attaches the provided tags to the graph immutably.
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/definitions/graph_definition.py:789]: Tangential to this PR but it's pretty rough that `tag` and `tags` are so different. Seems like this should be named `with_tag` or something.

**Review by smackesey — APPROVED** [review]
yes

File: python_modules/dagster/dagster/_core/definitions/graph_definition.py (hunk 3 above)
Comment (dpeng817) [reply python_modules/dagster/dagster/_core/definitions/graph_definition.py:789]: I agree :/

**Pull request merged**